staging/src/github.com/openshift/sdn/pkg/network/node/healthcheck.go

```diff
@@ -55,40 +67,36 @@ func runOVSHealthCheck(network, addr string, healthFn func() error) {
 		err = c.WaitForDisconnect()
 		utilruntime.HandleError(fmt.Errorf("SDN healthcheck disconnected from OVS server: %v", err))
 
+		// Only log the final "timed out waiting for condition" error
+		// message since we know it'll take some time for the OVS container
+		// to restart or recover; no point in spamming logs while it does.
 		err = utilwait.PollImmediate(100*time.Millisecond, ovsRecoveryTimeout, func() (bool, error) {
-			c, err := ovsclient.DialTimeout(network, addr, ovsDialTimeout)
-			if err != nil {
-				klog.V(2).Infof("SDN healthcheck unable to reconnect to OVS server: %v", err)
-				return false, nil
-			}
-			defer c.Close()
-			if err := c.Ping(); err != nil {
-				klog.V(2).Infof("SDN healthcheck unable to ping OVS server: %v", err)
+			// Log reconnect messages at a lower level since we know it'll take
+			// some time for the OVS container to restart
+			if dialErr, pingErr := dialAndPing(network, addr); dialErr != nil || pingErr != nil {
 				return false, nil
 			}
 			if err := healthFn(); err != nil {
-				return false, fmt.Errorf("OVS health check failed: %v", err)
+				return false, fmt.Errorf("OVS reinitialization required: %v", err)
 			}
 			return true, nil
 		})
 		if err != nil {
 			// If OVS restarts and our health check fails, we exit
 			// TODO: make openshift-sdn able to reconcile without a restart
-			klog.Fatalf("SDN healthcheck detected unhealthy OVS server, restarting: %v", err)
+			klog.Fatalf("SDN healthcheck detected OVS server change, restarting: %v", err)
```

Comment:
There are two different cases here: OVS restarted and destroyed part of the network so we have to restart to recreate it, or else OVS exited and we timed out waiting for it to restart. It might be nice to distinguish those.

Reply:
Those will be distinguished from the error returned by PollImmediate().  If it destroys part of the network we'll print "OVS reinitialization required", while if it exits and we timeout waiting for it to restart we'll get "timed out waiting for condition" from PollImmediate() internally.